persistence-api/src/main/java/io/stargate/db/cdc/DefaultCDCHealthChecker.java

```diff
@@ -0,0 +1,136 @@
+package io.stargate.db.cdc;
+
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.EWMA;
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+class DefaultCDCHealthChecker implements CDCHealthChecker {
+  private double errorRateThreshold;
+  private int minErrorsPerSecond;
+  static final int INTERVAL = 5;
+  static final long TICK_INTERVAL = TimeUnit.SECONDS.toNanos(INTERVAL);
+
+  private AutoTickEWMA successes;
+  private AutoTickEWMA errors;
+
+  /**
+   * Creates a new instance of {@link DefaultCDCHealthChecker}.
+   *
+   * @param errorRateThreshold The percentage of error requests compared to total, expressed in as a
+   *     double from 0 to 1.
+   * @param minErrorsPerSecond The minimum amount of error occurrences per second for the health
+   *     checker to consider the error ratio. This setting is created to prevent detecting the CDC
+   *     producer as unhealthy when there's low traffic and few errors.
+   * @param ewmaIntervalMinutes The interval to determine the coefficient for the degree of
+   *     weighting decrease in the exponentially weighted moving average (EWMA). The health checker
+   *     will use this value to set a soothing factor equivalent to UNIX load average.
```

Comment:
Typo?
```suggestion
   *     will use this value to set a smoothing factor equivalent to UNIX load average.
```

Reply:
hahaha, it can be a _soothing_ factor also :dadjoke:
good catch